core/src/main/java/hudson/security/HudsonPrivateSecurityRealm.java

```diff
@@ -717,14 +717,18 @@ public Details newInstance(StaplerRequest req, JSONObject formData) throws FormE
                 if(!Util.fixNull(pwd).equals(Util.fixNull(pwd2)))
                     throw new FormException("Please confirm the password by typing it twice","user.password2");
 
+                User user = Util.getNearestAncestorOfTypeOrThrow(req, User.class);
+                UserProperty p = user.getProperty(Details.class);
+
                 String data = Protector.unprotect(pwd);
                 if(data!=null) {
                     String prefix = Stapler.getCurrentRequest().getSession().getId() + ':';
                     if(data.startsWith(prefix))
+                        SecurityListener.fireUserPropertyUpdated(p, user.getId());
```

Comment:
⚠️ missing `{}` in the `if`, so you are breaking the statement there.

That's exactly the main reason why I *hate* the `if` without brackets 👿 

If that code bug was not discovered by your tests => there is a missing path in your coverage.

Reply:
do you advise using the `if` for only one-liners; or to just never use a non-bracketed `if` at all ?

Like;
```
                if((data!=null) && (p != null)) {
                    String prefix = Stapler.getCurrentRequest().getSession().getId() + ':';
                    if(data.startsWith(prefix)) return Details.fromHashedPassword(data.substring(prefix.length()));
                }
```

versus;

```                if((data!=null) && (p != null)) {
                    String prefix = Stapler.getCurrentRequest().getSession().getId() + ':';
                    if(data.startsWith(prefix)) {
                        return Details.fromHashedPassword(data.substring(prefix.length()));
                    }
                }
```

Reply:
I can only advice to never use a `if` without the associated `{}`, to avoid situation where someone wants to add a line, but due to the missing brackets will break the code flow like here.

And the one-liner is worse from my PoV due to the poor readability.

Reply:
perks of using Python over Java :)

Reply:
I only use one line ifs for things like:

```if (foo == null) return null;```

or

```if (bar) return false;```

or

```if (baz) return;```

Basically, only for control flow type things. Remember, forgetting your brackets can cause security problems (see the old Apple bug). If you auto-format your code before committing, you can see if you missed brackets based on how it indents things.